services/web/server/src/simcore_service_webserver/utils_aiohttp.py

```diff
@@ -128,3 +129,38 @@ class NextPage(BaseModel, Generic[PageParameters]):
         ..., description="Code name to the front-end page. Ideally a PageStr"
     )
     parameters: PageParameters | None = None
+
+
+def iter_origins(request: web.Request) -> Iterator[tuple[str, str]]:
+    #
+    # SEE https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-Forwarded-Host
+    # SEE https://doc.traefik.io/traefik/getting-started/faq/#what-are-the-forwarded-headers-when-proxying-http-requests
+    seen = set()
+
+    # X-Forwarded-Proto and X-Forwarded-Host can contain a comma-separated list of protocols and hosts
+    # (when the request passes through multiple proxies)fwd_protos = [
+    fwd_protos = [
+        p.strip()
+        for p in request.headers.get("X-Forwarded-Proto", "").split(",")
+        if p.strip()
+    ]
+    fwd_hosts = [
+        h.strip()
+        for h in request.headers.get("X-Forwarded-Host", "").split(",")
+        if h.strip()
+    ]
+
+    if fwd_protos and fwd_hosts:
+        for proto, host in zip(fwd_protos, fwd_hosts, strict=False):
+            if (proto, host) not in seen:
+                seen.add((proto, host))
+                yield (proto, host.partition(":")[0])  # strip port
+
+    # fallback to request scheme/host
+    yield request.scheme, f"{request.host.partition(':')[0]}"
+
+
+def get_api_base_url(request: web.Request) -> str:
+    scheme, host = next(iter_origins(request))
+    api_host = f"api.{host}" if not is_ip_address(host) else host
```

Comment:
should be more redable flipped
```suggestion
    api_host = host if is_ip_address(host) else f"api.{host}"
```